db: TestIteratorErrors error swallowed

```
    external_test.go:126: iter38.First() // [false] <nil> #4429
    external_test.go:90: last injected error: injected error
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/pebble/vfs/errorfs.LabelledError.MaybeError
          | 	/Users/runner/work/pebble/pebble/vfs/errorfs/dsl.go:239
          | github.com/cockroachdb/pebble/vfs/errorfs.(*Counter).MaybeError
          | 	/Users/runner/work/pebble/pebble/vfs/errorfs/errorfs.go:217
          | github.com/cockroachdb/pebble/vfs/errorfs.(*Toggle).MaybeError
          | 	/Users/runner/work/pebble/pebble/vfs/errorfs/errorfs.go:259
          | github.com/cockroachdb/pebble/vfs/errorfs.(*errorFile).ReadAt
          | 	/Users/runner/work/pebble/pebble/vfs/errorfs/errorfs.go:482
          | github.com/cockroachdb/pebble/objstorage/objstorageprovider.(*vfsReadHandle).ReadAt
          | 	/Users/runner/work/pebble/pebble/objstorage/objstorageprovider/vfs_readable.go:163
          | github.com/cockroachdb/pebble/sstable/block.(*Reader).doRead
          | 	/Users/runner/work/pebble/pebble/sstable/block/block.go:545
          | github.com/cockroachdb/pebble/sstable/block.(*Reader).Read
          | 	/Users/runner/work/pebble/pebble/sstable/block/block.go:501
          | github.com/cockroachdb/pebble/sstable.(*Reader).readDataBlock
          | 	/Users/runner/work/pebble/pebble/sstable/reader.go:372
          | github.com/cockroachdb/pebble/sstable.(*singleLevelIterator[...]).loadDataBlock
          | 	/Users/runner/work/pebble/pebble/sstable/reader_iter_single_lvl.go:493
          | github.com/cockroachdb/pebble/sstable.(*singleLevelIterator[...]).NextPrefix
          | 	/Users/runner/work/pebble/pebble/sstable/reader_iter_single_lvl.go:1293
          | github.com/cockroachdb/pebble/internal/keyspan.(*InterleavingIter).NextPrefix
          | 	/Users/runner/work/pebble/pebble/internal/keyspan/interleaving_iter.go:494
          | github.com/cockroachdb/pebble.(*levelIter).NextPrefix
          | 	/Users/runner/work/pebble/pebble/level_iter.go:764
          | github.com/cockroachdb/pebble.(*mergingIter).nextEntry
          | 	/Users/runner/work/pebble/pebble/merging_iter.go:543
          | github.com/cockroachdb/pebble.(*mergingIter).NextPrefix
          | 	/Users/runner/work/pebble/pebble/merging_iter.go:1260
          | github.com/cockroachdb/pebble.(*Iterator).internalNextPrefix
          | 	/Users/runner/work/pebble/pebble/iterator.go:1882
          | github.com/cockroachdb/pebble.(*Iterator).nextPrefix
          | 	/Users/runner/work/pebble/pebble/iterator.go:1774
          | github.com/cockroachdb/pebble.(*Iterator).NextPrefix
          | 	/Users/runner/work/pebble/pebble/iterator.go:1740
          | github.com/cockroachdb/pebble/metamorphic.(*retryableIter).NextPrefix.func1
          | 	/Users/runner/work/pebble/pebble/metamorphic/retryable.go:124
          | github.com/cockroachdb/pebble/metamorphic.(*retryableIter).withRetry
          | 	/Users/runner/work/pebble/pebble/metamorphic/retryable.go:47
          | github.com/cockroachdb/pebble/metamorphic.(*retryableIter).NextPrefix
          | 	/Users/runner/work/pebble/pebble/metamorphic/retryable.go:123
          | github.com/cockroachdb/pebble/metamorphic.(*iterNextPrefixOp).run
          | 	/Users/runner/work/pebble/pebble/metamorphic/ops.go:1723
          | github.com/cockroachdb/pebble/metamorphic.(*Test).runOp
          | 	/Users/runner/work/pebble/pebble/metamorphic/test.go:440
          | github.com/cockroachdb/pebble/metamorphic.(*Test).step
          | 	/Users/runner/work/pebble/pebble/metamorphic/test.go:416
          | github.com/cockroachdb/pebble/metamorphic.(*Test).Step
          | 	/Users/runner/work/pebble/pebble/metamorphic/test.go:405
          | github.com/cockroachdb/pebble_test.TestIteratorErrors
          | 	/Users/runner/work/pebble/pebble/external_test.go:102
          | testing.tRunner
          | 	/Users/runner/hostedtoolcache/go/1.24.4/arm64/src/testing/testing.go:1792
          | runtime.goexit
          | 	/Users/runner/hostedtoolcache/go/1.24.4/arm64/src/runtime/asm_arm64.s:1223
        Wraps: (2) injected error
        Error types: (1) *withstack.withStack (2) errorfs.LabelledError
panic: pebble: invariant violation: accumulated error swallowed [recovered]
	panic: pebble: invariant violation: accumulated error swallowed [recovered]
	panic: pebble: invariant violation: accumulated error swallowed
```
https://github.com/cockroachdb/pebble/actions/runs/16055511369/job/45308785077?pr=5025

Jira issue: PEBBLE-539

--- a/internal/keyspan/interleaving_iter.go
+++ b/internal/keyspan/interleaving_iter.go
@@ -492,7 +492,9 @@ func (i *InterleavingIter) NextPrefix(succKey []byte) *base.InternalKV {
 		return nil
 	case posPointKey:
 		i.savePoint(i.pointIter.NextPrefix(succKey))
-		if i.withinSpan {
+		if i.err != nil {
+			i.pos = posExhausted
+		} else if i.withinSpan {
 			if i.pointKV == nil || i.cmp(i.span.End, i.pointKV.K.UserKey) <= 0 {
 				i.pos = posKeyspanEnd
 			} else {

